library/tasks/HISAT2.wdl

```diff
@@ -13,6 +13,7 @@ task HISAT2PairedEnd {
   # use provided disk number or dynamically size on our own, 10 is our zipped fastq -> bam conversion with 50GB of additional disk
   Int disk = ceil((size(fastq1, "GB") + size(fastq2, "GB") * 10) + size(hisat2_ref, "GB") + 50)
   Int preemptible = 5
+  Int max_retries = 0
```

Comment:
I'm not sure if understand why we set 0 here. I guess the upper level(workflow level) matx_retries will set it to 1 if it's available, but it will be unclear to me whether this value is 0 or 1 if we set values at both places. So how about using something like (I'm not sure if the following syntax is correct):
```
Int max_retries = select_first([max_retries, 0])
```
So it guarantees to attempt to use the workflow value first.

Reply:
This is already the way Cromwell assigns variables. The priority order is:
- use value from inputs json
- use default value set at workflow level
- use default value set at task level

So setting zero here means we use zero as the default but allow it to be overridden. This is the same way we're setting default values for other task parameters like preemptibles and cpu.

Reply:
I see I'm not familiar with this mechanism, thanks for the explanation here!

Reply:
There are a variety of use cases that I'm trying to keep in mind when setting the retries value. I think we should only have retries turned on in HCA production and scale testing. If someone is running this workflow outside of those use cases, they are either:
1. Testing out modifications - retries should be off because the changes are experimental with a good chance of failure, so retrying is a waste of time and resources
2. Reusing it for production outside HCA. Retries might make sense here but turning it off by default makes sense -- you have to opt into it deliberately if you want it on.

Similarly, someone might reuse individual tasks when testing out changes or building their own pipeline and I think it makes sense to have retries turned off by default in those situations, too.

The way we're setting the variables, we will only have it turned on when Lira launches the workflows (where it will set max retries as an input parameter), which will avoid having it turned on in most cases outside of production and scale testing.

Reply:
I haven't thought these use cases, this explanation totally makes sense to me. Thanks!